vms/platformvm/state/state.go

```diff
@@ -348,13 +348,12 @@ func newState(
 		singletonDB:            singletonDB,
 		baseMerkleDB:           baseMerkleDB,
 		merkleDB:               merkleDB,
-		changesSinceLastCommit: make([]database.BatchOp, 0),
+		changesSinceLastCommit: make(map[string]database.BatchOp),
 
 		currentStakers: newBaseStakers(),
 		pendingStakers: newBaseStakers(),
 
-		delegateeRewardCache:     make(map[ids.ID]map[ids.NodeID]uint64),
-		modifiedDelegateeRewards: make(map[ids.ID]map[ids.NodeID]uint64),
```

Comment:
This is the idea: keep all stuff to serve queries in *cache, including diff just added.
Keep diff in a serialized state already in changesSinceLastCommit, so that it's ready to be applied or calculate view